src/shim/state_tree.rs

```diff
@@ -94,6 +94,7 @@ impl TryFrom<StateTreeVersion> for StateTreeVersionV3 {
 /// Not all the inner methods are implemented, only those that are needed. Feel
 /// free to add those when necessary.
 pub enum StateTree<S> {
```

Comment:
To keep the code consistent, you must distinguish between state-tree and FVM versions. Each FVM state-tree structure covers a range of state-tree versions.